tests/test_dcrAssembleCoadd.py

```diff
@@ -0,0 +1,295 @@
+        azimuth = 130.*afwGeom.degrees
+        elevation = 70.*afwGeom.degrees
+        visitInfo = self.makeDummyVisitInfo(azimuth, elevation)
+        wcs = self.makeDummyWcs(cdRotAngle, visitInfo=visitInfo)
+        rotAngle = DcrAssembleCoaddTask.calculateRotationAngle(visitInfo, wcs)
+        refAngle = Angle(-0.9344289857053072)
+        self.assertAnglesNearlyEqual(refAngle, rotAngle)
+
+    def testConditionDcrModelNoChange(self):
+        """! Conditioning should not change the model if it is identical to the reference.
+
+        This additionally tests that the variance and mask planes do not change.
+        """
+        refModels = [model.clone() for model in self.dcrModels]
+        DcrAssembleCoaddTask.conditionDcrModel(refModels, self.dcrModels, self.bbox, gain=1.)
+        for model, refModel in zip(self.dcrModels, refModels):
+            self.assertMaskedImagesEqual(model, refModel)
+
+    def testConditionDcrModelWithChange(self):
+        """! Verify the effect of conditioning when the model changes by a known amount.
+
+        This additionally tests that the variance and mask planes do not change.
+        """
+        refModels = [model.clone() for model in self.dcrModels]
+        for model in self.dcrModels:
+            model.getImage().getArray()[:, :] *= 3.
+        DcrAssembleCoaddTask.conditionDcrModel(refModels, self.dcrModels, self.bbox, gain=1.)
+        for model, refModel in zip(self.dcrModels, refModels):
+            refModel.getImage().getArray()[:, :] *= 2.
+            self.assertMaskedImagesEqual(model, refModel)
+
+    def testShiftImagePlane(self):
+        """! Verify the shift calculation for the image and variance planes.
+
+        The shift of the mask plane is tested separately since it is calculated differently.
+        """
+        rotAngle = Angle(0.)
+        azimuth = 200.*afwGeom.degrees
+        elevation = 75.*afwGeom.degrees
+        visitInfo = self.makeDummyVisitInfo(azimuth, elevation)
+        wcs = self.makeDummyWcs(rotAngle, visitInfo=visitInfo)
+        dcrShift = DcrAssembleCoaddTask.dcrShiftCalculate(visitInfo, wcs,
+                                                          self.config.lambdaEff,
+                                                          self.config.filterWidth,
+                                                          self.config.dcrNSubbands)
+        newMaskedImage = DcrAssembleCoaddTask.convolveDcrModelPlane(self.dcrModels[0], dcrShift[0],
+                                                                    bbox=self.bbox,
+                                                                    useFFT=False,
+                                                                    useInverse=False)
+        shift = (dcrShift[0].dy, dcrShift[0].dx)
+        refImage = scipy.ndimage.interpolation.shift(self.dcrModels[0].getImage().getArray(), shift)
+        refVariance = scipy.ndimage.interpolation.shift(self.dcrModels[0].getVariance().getArray(), shift)
```

Comment:
All you're doing is rewriting the code in `DcrAssembleCoaddTask.convolveDcrModelPlane` --- that's not a real test. You should do a proper test: maybe centroid the sources on the model images to verify the shifts?